### ISSUE TITLE
Exeption using Django+channels ASGI

### ISSUE THREAD

**Issue by hubertokf** [getsentry/sentry-python#911]
When I use Django + Channels and ASGI application I got this exception.

```
Exception inside application: sentry_patched_asgi_handler() takes 3 positional arguments but 4 were given
Traceback (most recent call last):
  File "env/lib/python3.8/site-packages/channels/routing.py", line 71, in __call__
    return await application(scope, receive, send)
TypeError: sentry_patched_asgi_handler() takes 3 positional arguments but 4 were given
HTTP POST /api/v2/auth/login 500 [0.67, 127.0.0.1:37750]
```

This is my sdk init:

```
sentry_sdk.init(
    dsn="...",
    integrations=[DjangoIntegration()],
    environment=os.getenv('ENVIRONMENT', "development"),

    # If you wish to associate users to errors (assuming you are using
    # django.contrib.auth) you may enable sending PII data.
    send_default_pii=True
)
```

**Comment by untitaker** [comment]
Can you tell me which versions of channels, django etc you have?

**Comment by hubertokf** [comment]
Sure! This is my init log:

```
System check identified no issues (0 silenced).
November 04, 2020 - 13:55:24
Django version 2.2.11, using settings 'prendaApi.settings'
Starting ASGI/Channels version 3.0.0 development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
```

**Comment by hubertokf** [comment]
I have just updated to `sentry-sdk==0.19.2` and the Exception persists.

**Comment by lpomfrey** [comment]
We're experiencing the same problem, downgrading to channels < 3.0 fixes the issue. Looks like https://github.com/django/channels/commit/4fe2b81df0fa8b54df19054d78a17fd2d28da2a9 is the relevant change.

**Comment by untitaker** [comment]
Thanks @lpomfrey this helps a lot!

**Comment by lpomfrey** [comment]
With a little further investigation it seems that for channels 3 `patch_channels_asgi_handler_impl` from https://github.com/getsentry/sentry-python/blob/549b7df3707cb41edf88390a75132434d3ed8c01/sentry_sdk/integrations/django/asgi.py#L65 should do the same as `patch_django_asgi_handler_impl` from https://github.com/getsentry/sentry-python/blob/549b7df3707cb41edf88390a75132434d3ed8c01/sentry_sdk/integrations/django/asgi.py#L33

**Comment by untitaker** [comment]
Yes exactly. Since you're investigating, do you have time to make this patch?

**Comment by lpomfrey** [comment]
Yes, not a problem.

**Comment by casper936** [comment]
I have the same problem when I use Django + Channels

```Traceback (most recent call last):
  File "/home/casper/PycharmProjects/s7/epaas_forms/.venv/lib/python3.7/site-packages/channels/routing.py", line 71, in __call__
    return await application(scope, receive, send)
TypeError: sentry_patched_asgi_handler() takes 3 positional arguments but 4 were given
2020-11-11 18:48:19,505 -  ERROR - django.channels.server - HTTP GET / 500 [0.03, 127.0.0.1:33742]
2020-11-11 18:48:20,559 -  ERROR - daphne.server - Exception inside application: sentry_patched_asgi_handler() takes 3 positional arguments but 4 were given
```

Init log:
```System check identified no issues (0 silenced).
November 11, 2020 - 18:48:10
Django version 3.1.3, using settings 'forms.settings'
Starting ASGI/Channels version 3.0.2 development server at http://localhost:8000/
Quit the server with CONTROL-C.
```

sentry-sdk==0.19.2

**Comment by untitaker** [comment]
@casper936 the problem is fixed in master and will be fixed in 0.19.3

**Comment by dkobia** [comment]
Confirmed that the issue is fixed. Currently using `0.20.3`.
